common4j/src/main/com/microsoft/identity/common/java/broker/ICallValidator.java

```diff
@@ -0,0 +1,46 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.java.broker;
+
+import com.microsoft.identity.common.java.exception.ClientException;
+
+import java.util.Map;
+
+import lombok.NonNull;
+
+/**
+ * An interface that throws an exception if the caller is not acceptable given a UID.
+ */
+public interface ICallValidator {
+    /**
+     * Throws a ClientException if the caller cannot be validated or is unauthorized.
+     * @param methodName the method name for logging purposes.
+     * @param callingUid the identifier for the caller, platform dependent.
+     * @param allowedApplications A map of name to iterable of verification string for calling apps.
```

Comment:
What are these Strings representing? client id?

Reply:
In android, they are the application identifier and the hash.  There's no reason that they need to be the same thing in Linux.

Reply:
done